standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -314,6 +314,8 @@ abstract class TxnHandler implements TxnStore, TxnStore.MutexAPI {
   private static final String  EXCL_CTAS_ERR_MSG = 
       "Failed to initiate a concurrent CTAS operation with the same table name, lockInfo : %s";
   private static final String ZERO_WAIT_READ_ERR_MSG = "Unable to acquire read lock due to an existing exclusive lock {%s}";
+  private static String SHOW_COMPACTION_ORDERBY_CLAUSE = "  order by  case when CC_END > CC_START and CC_END > CC_ENQUEUE_TIME then CC_END When CC_START > CC_ENQUEUE_TIME then CC_START Else CC_ENQUEUE_TIME end desc";
```

Comment:
CC_ENQUEUE_TIME - inserted in a queue;
CC_START - worker started;
CC_COMMIT_TIME - end of worker/compaction
CQ_CLEANER_START - cleaner started
CC_END - end of cleaner

shouldn't it be something like:
````
order by 
case 
when CC_END > CC_START and CC_END > CQ_CLEANER_START 
	then CC_END 
when CC_START > CQ_CLEANER_START 
	then CC_START 
	else CQ_CLEANER_START 
end desc, 
CC_ENQUEUE_TIME asc;
````

Reply:
resolved